pkg/cliplugins/workspace/plugin/kubeconfig.go

```diff
@@ -215,8 +215,21 @@ func (o *UseWorkspaceOptions) Run(ctx context.Context) error {
 		if err != nil {
 			return err
 		}
-		newServerHost = homeWorkspace.Spec.URL
-
+		u, homeCluster, err := pluginhelpers.ParseClusterURL(homeWorkspace.Spec.URL)
```

Comment:
`homeWorkspace.Spec.URL` has the path. We just have to extract it from the URL. We have a helper here somewhere.

Reply:
Would rather not go through the `LogicalCluster` for that.

Reply:
I do not think currently the value lf homeWorkspace.Spec.URL is the same as path annotation in `LogicalCluster`. `homeWorkspace.Spec.URL` is `https://xxx/clusters/kvdk2spgmbix `  while path annotation is `user:kcp-admin`. Should `homeWorkspace.Spec.URL`  be `https://xxx/clusters/user:kcp-admin`?

Reply:
> Would rather not go through the LogicalCluster for that.

@sttts what would be an alternate solution ?

Reply:
> Should homeWorkspace.Spec.URL be https://xxx/clusters/user:kcp-admin?

Yes, I think it should.

Reply:
Ah, so that means that the problem is at a different level.

Reply:
@sttts would you check if https://github.com/kcp-dev/kcp/pull/2846 is the appropriate way.